hooks/post_gen_project.py

```diff
@@ -502,8 +504,131 @@ def main():
 
 
+def manipulate_pyproject():
+
+    path = Path("pyproject.toml")
+    project = parse(path.read_text())
+
+    if "{{cookiecutter.frontend_pipeline}}" != "Gulp":
+        del project["tool"]["pytest"]["ini_options"]["norecursedirs"]
+
+    if "{{cookiecutter.use_drf}}" == "y":
+        project["tool"]["mypy"]["plugins"].append("mypy_drf_plugin.main")
+        remove_dependency(project, "djangorestframework")
+        remove_dependency(project, "django-cors-headers")
+        remove_dependency(project, "drf-spectacular")
+
+    if "{{cookiecutter.use_drf}}" == "n":
+        remove_dev_dependency(project, "djangorestframework-stubs")
+
+    if "{{cookiecutter.use_docker}}" == "y":
+        replace_dependency(project, "psycopg", "psycopg[c]")
+    else:
+        replace_dependency(project, "psycopg", "psycopg[binary]")
+
+    if "{{cookiecutter.use_async}}" == "n" and "{{cookiecutter.use_async}}" == "n":
+        remove_dev_dependency(project, "watchfiles")
+
+    if "{{cookiecutter.frontend_pipeline}}" != "Django Compressor":
+        remove_dependency(project, "rcssmin")
+        remove_dependency(project, "django-compressor")
+
+    if "{{cookiecutter.frontend_pipeline}}" != "Webpack":
+        remove_dependency(project, "django-webpack-loader")
+
+    if "{{cookiecutter.use_whitenoise}}" != "y":
+        remove_dependency(project, "whitenoise")
+
+    if "{{cookiecutter.use_whitenoise}}" != "n":
+        remove_dependency(project, "Collectfasta")
+
+    if "{{cookiecutter.use_sentry}}" != "y":
+        remove_dependency(project, "sentry-sdk")
+
+    if "{{cookiecutter.use_docker}}" != "n" or "{{cookiecutter.windows}}" != "y":
+        remove_dependency(project, "hiredis")
+
+    if "{{cookiecutter.use_celery}}" != "y":
+        remove_dev_dependency(project, "celery")
+        remove_dev_dependency(project, "django-celery-beat")
+
+        if "{{cookiecutter.use_docker}}" != "y":
+            remove_dev_dependency(project, "flower")
+
+    if "{{cookiecutter.use_async}}" != "y":
+        remove_dependency(project, "uvicorn[standard]")
+        remove_dependency(project, "uvicorn-worker")
+
+    if "{{cookiecutter.cloud_provider}}":
+        map = {
+            "AWS": "s3",
+            "GCP": "google",
+            "Azure": "azure",
+        }
+
+        p = map.get("{{cookiecutter.cloud_provider}}", None)
+        if p:
+            replace_dependency(project, "django-storages", f"django-storages[{p}]")
+        else:
+            remove_dependency(project, "django-storages")
+
+    if "{{cookiecutter.mail_service}}":
+        map = {
+            "Mailgun": "mailgun",
+            "Amazon SES": "amazon-ses",
+            "Mailjet": "mailjet",
+            "Mandrill": "mandrill",
+            "Postmark": "postmark",
+            "Sendgrid": "sendgrid",
+            "Brevo": "brevo",
+            "SparkPost": "sparkpost",
+            "Other SMTP": "",
+        }
+
+        p = map.get("{{cookiecutter.mail_service}}", None)
```

Comment:
Are we expecting all options to be in the above map? If yes, maybe we should do `map["{{cookiecutter.mail_service}}"]` to make it error loudly if one is missing. I'm worried that we might forget this if we add another provider one day